### PR TITLE
[diffusion] [NPU] support ring attention on NPU with FIA attention backend

### PR DESCRIPTION
**Pull request by Makcum888e**
## Motivation

`ring attention` requires `return_softmax_lse` but `sdpa backend` cannot support this option. Support of `ring attention` on NPU can be implemented using `npu_fused_infer_attention_score`.

## Modifications

added new attention backend for NPU - FIA
added new function to platform interface `default_ring_attention_backend` to allow the platform to select the default backend

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Copied and adapted from: https://github.com/hao-ai-lab/FastVideo
+
+# SPDX-License-Identifier: Apache-2.0
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: please check the copyright header, do we really `copied and adapted from ...`?

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import torch
+
+from sglang.multimodal_gen.runtime.layers.attention.backends.attention_backend import (  # FlashAttentionMetadata,
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: please delete useless code

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        causal: bool,
+        softmax_scale: float,
+        num_kv_heads: int | None = None,
+        prefix: str = "",
+        **extra_impl_args,
+    ) -> None:
+        self.causal = causal
+        self.softmax_scale = softmax_scale
+        self.num_heads = num_heads
+        self.num_kv_heads = num_kv_heads if num_kv_heads is not None else num_heads
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: `self.num_kv_heads = num_kv_heads or num_heads`

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py (hunk 1 above)
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: yes, you are right, removed

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py (hunk 2 above)
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: done

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py (hunk 3 above)
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/fia.py]: done

**Review by BBuf — APPROVED** [review]
